From 164d1969018d64e10fb58bf2954c61d171774c6a Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Thu, 13 Aug 2015 07:23:09 -0700
Subject: [PATCH 230/827] remoteproc: zynqmp_r5_remoteproc: remove unused vring0 mem resource

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Remove memory resource vring0 define in DTS and remove getting vring0
memory resource from probe().

Signed-off-by: Wendy Liang <jliang@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 2e3e360b7f9979be139ad6bd4e477d75ec766b10)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt | 4 ++--
 drivers/remoteproc/zynqmp_r5_remoteproc.c                      | 7 -------
 2 files changed, 2 insertions(+), 9 deletions(-)

diff --git a/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt b/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt
index 2b9cc16..f3b92c6 100644
--- a/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt
+++ b/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt
@@ -31,8 +31,8 @@ Example:
 --------
 	zynqmp-r5-remoteproc@0 {
 		compatible = "xlnx,zynqmp-r5-remoteproc";
-		reg = <0x0 0x3ed00000 0x800000>, <0x0 0xff300000 0x100>, <0x0 0xff9a0000 0x400>, <0x0 0xff5e0000 0x400>;
-		reg-names = "vring0", "ipi", "rpu_base", "apb_base";
+		reg = <0x0 0xff300000 0x100>, <0x0 0xff9a0000 0x400>, <0x0 0xff5e0000 0x400>;
+		reg-names = "ipi", "rpu_base", "apb_base";
 		core_conf = "split0";
 		method = "direct";
 		interrupt-parent = <&gic>;
diff --git a/drivers/remoteproc/zynqmp_r5_remoteproc.c b/drivers/remoteproc/zynqmp_r5_remoteproc.c
index c41dc2d..6b594ff 100644
--- a/drivers/remoteproc/zynqmp_r5_remoteproc.c
+++ b/drivers/remoteproc/zynqmp_r5_remoteproc.c
@@ -579,13 +579,6 @@ static int zynqmp_r5_remoteproc_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, local);
 
-	/* Declare vring for firmware */
-	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "vring0");
-	if (!res) {
-		dev_err(&pdev->dev, "invalid address for vring0\n");
-		return -ENXIO;
-	}
-
 	/* FIXME: it may need to extend to 64/48 bit */
 	ret = dma_set_coherent_mask(&pdev->dev, DMA_BIT_MASK(32));
 	if (ret) {
-- 
2.9.3

